src/mrpro/operators/Functional.py

```diff
@@ -173,3 +187,80 @@ class ElementaryProximableFunctional(ElementaryFunctional, ProximableFunctional)
     A proximable functional is a functional :math:`f(x)` that has a prox implementation,
     i.e. a function that yields :math:`argmin_x \sigma f(x) + 1/2 \|x - y\|^2`.
     """
+
+
+class ScaledFunctional(Functional):
+    """Functional scaled by a scalar."""
+
+    def __init__(self, functional: Functional, scale: torch.Tensor | complex) -> None:
```

Comment:
similarly here, I don't understand why we give the option of complex if in the parameters we write that it must be real and positive?

Reply:
or is this a constraint only for the prox to be well-defined? In that case I would write it separately in the description of ScaledFunctional and ScaledProximableFunctional

Reply:
Good catch.
The reason is that all ProximalFunctionals are Operators.
We can multiply all Operators with complex numbers.
Thus, from a type hinting perspective, we must also allow multiplication of ProximableFunctional with complex.
I will change it to make it consistent complex everywhere.
In the end, the code will error if you call .prox on something that after applying all scalings etc is still complex or negative.

So (1j*(-1j)*Functional).prox() would work, but (1j*Functional).prox() would fail.

Does that make sense?